src/ItemList.js

```diff
@@ -1,25 +1,82 @@
 import React, { useState, useEffect } from 'react';
 import Item from './Item';
 import { useHistory } from 'react-router-dom';
+import filter from './lib/filter';
+import { differenceInDays, fromUnixTime } from 'date-fns';
 
 function ItemList(props) {
   const [query, setQuery] = useState('');
-  const [queryArray, setQueryArray] = useState([]);
+  const [queryObj, setQueryObj] = useState({
+    week: [],
+    month: [],
+    longer: [],
+    inactive: [],
+  });
   let history = useHistory();
   const redirect = () => {
     history.push('/additems');
   };
 
   useEffect(() => {
-    // [filter-list] 2. Comparison function to filter shopping list and create a search results array
-    // [filter-list] (Remove accidental space character from query, make lowercase, and prevent regex errors)
-    const resultsArray = props.list.filter((itemObj) => {
-      return itemObj['itemName']
-        .toLowerCase()
-        .replace(/[.*+?^${}()|[\]\\]/g, '\\$&')
-        .includes(query.toLowerCase().trim());
+    const resultsObj = {
+      week: [],
+      month: [],
+      longer: [],
+      inactive: [],
+    };
+
+    props.list.forEach((item) => {
+      if (item.purchaseDates.length < 1 || item.purchaseEstimates.length < 1) {
```

Comment:
To help with backwards compatibility, there are some items which do not have the `purchaseEstimates` field so getting the length with throw and error. You can fix this pretty easily in this conditional:

```suggestion
      if (item.purchaseDates.length < 1 || !(item.purchaseEstimates?.length > 0)) {
```

Reply:
This is a much more elegant way of handling empty fields. Thanks, @luisaugusto!